exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/DateTypeFunctions.java

```diff
@@ -425,43 +425,73 @@ public void eval() {
         @Param TimeStampHolder left;
         @Param TimeStampHolder right;
         @Output IntervalHolder out;
+        @Inject ContextInformation contextInfo;
 
         @Override
         public void setup() {
         }
 
         @Override
         public void eval() {
-            long diff = left.value - right.value;
-            long days = diff / org.apache.drill.exec.vector.DateUtilities.daysToStandardMillis;
-            out.months = (int) (days / org.apache.drill.exec.vector.DateUtilities.monthToStandardDays);
-            out.days = (int) (days % org.apache.drill.exec.vector.DateUtilities.monthToStandardDays);
-            out.milliseconds = (int) (diff % org.apache.drill.exec.vector.DateUtilities.daysToStandardMillis);
+            java.time.OffsetDateTime from = java.time.Instant.ofEpochMilli(right.value).atOffset(java.time.ZoneOffset.UTC);
+            java.time.OffsetDateTime to = java.time.Instant.ofEpochMilli(left.value).atOffset(java.time.ZoneOffset.UTC);
+            java.time.Duration duration = java.time.Duration.between(from.toLocalTime(), to.toLocalTime());
+            java.time.Period period;
+
+            if (from.isAfter(to) && duration.compareTo(java.time.Duration.ZERO) > 0) {
```

Comment:
This code appears to be a copy/paste of the code below (or visa versa.) In general, reuse via copy/past is discouraged.

See note below about factoring out the (common) code into a separate function which you can test and call from both of the UDFs.

Note that I'd suggest doing the Drill-type-to-Java type conversions in that function. These conversions are complex; it would be good to have extensive unit tests. Running queries to test the code is, as noted, below, too cumbersome.

Reply:
The implementation has been simplified, is it necessary to separate it into a separate method to remove copy/past?